tools/server-admin/src/main/java/org/projectnessie/tools/admin/cli/ExpireSnapshotTasks.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright (C) 2024 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.tools.admin.cli;
+
+import static org.projectnessie.versioned.storage.common.logic.Logics.commitLogic;
+import static org.projectnessie.versioned.storage.common.logic.Logics.indexesLogic;
+import static org.projectnessie.versioned.storage.versionstore.RefMapping.hashNotFound;
+import static org.projectnessie.versioned.storage.versionstore.TypeMapping.hashToObjId;
+import static org.projectnessie.versioned.storage.versionstore.TypeMapping.storeKeyToKey;
+
+import com.google.common.collect.Iterators;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Spliterators;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+import org.projectnessie.catalog.service.objtypes.EntitySnapshotObj;
+import org.projectnessie.model.Content;
+import org.projectnessie.model.ContentKey;
+import org.projectnessie.model.IcebergTable;
+import org.projectnessie.model.IcebergView;
+import org.projectnessie.versioned.Hash;
+import org.projectnessie.versioned.ReferenceNotFoundException;
+import org.projectnessie.versioned.storage.common.exceptions.ObjNotFoundException;
+import org.projectnessie.versioned.storage.common.indexes.StoreIndex;
+import org.projectnessie.versioned.storage.common.objtypes.CommitObj;
+import org.projectnessie.versioned.storage.common.objtypes.CommitOp;
+import org.projectnessie.versioned.storage.common.persist.ObjId;
+import org.projectnessie.versioned.storage.versionstore.ContentMapping;
+import picocli.CommandLine;
+
+@CommandLine.Command(
+    name = "expire-snapshot-tasks",
+    mixinStandardHelpOptions = true,
+    description =
+        "Delete all state of Iceberg snapshot loading tasks previously executed by the Nessie Catalog.")
+public class ExpireSnapshotTasks extends BaseCommand {
```

Comment:
Imho the name is a bit misleading. Initially I thought this would run some GC expire-snapshot job. Maybe `DeleteIcebergTasks` or `DeleteCatalogTasks`?

Reply:
I'll rename to `DeleteCatalogTasks`

Reply:
renamed